connectors/airwatch/src/main/java/com/vmware/connectors/airwatch/AirWatchController.java

```diff
@@ -115,8 +118,18 @@ public AirWatchController(AsyncRestOperations rest, CardTextAccessor cardTextAcc
 
         String udid = cardRequest.getTokenSingleValue(UDID_KEY);
         String clientPlatform = cardRequest.getTokenSingleValue(PLATFORM_KEY);
+
+        if (StringUtils.isAnyBlank(udid, clientPlatform)) {
+            logger.debug("Either device UDID or client platform is blank.");
+            return Single.just(new ResponseEntity<>(BAD_REQUEST));
+        }
+
         Set<String> appKeywords = cardRequest.getTokens("app_keywords");
 
+        if (CollectionUtils.isEmpty(appKeywords)) {
+            return Single.just(ResponseEntity.ok(new Cards()));
+        }
+
         HttpHeaders headers = new HttpHeaders();
         headers.set(AUTHORIZATION, awAuth);
 
```

Comment:
In line 161, we are fetching the authorization token by splitting the string based on ```(?i)Bearer```. Incase, if the client send an authorization token starts with ```Basic```, then it will result in ```ArrayIndexOutOfBoundsException```. 

We can avoid this by checking whether the ```authToken``` begins with ```Baisc`` or ```Bearer``` and we can retrieve the token accordingly. If it is not any of the above values, throw an ```InvalidAuthException``` back to the client.

Reply:
But since we support only Bearer type, this will result in 401.